packages/wp-checkout/src/lib/format-monetary-amount-for-locale.js

```diff
@@ -0,0 +1,781 @@
+				[ ( absoluteAmount % 1000 ).toString().padStart( 3, '0' ) ],
+				2
+			);
+
+		// Locales with "3*" style digit groups
+		default:
+			return groupDigitsAccum( absoluteAmount, [], 3 );
+	}
+}
+
+/**
+ * Number of minor units per major unit. When expressed as a power of 10 this is called the exponent.
+ * Some currencies have non-integer exponents (e.g. Madagascar) so we return the raw number instead.
+ *
+ * @param {string} currencyCode ISO 4217 currency code
+ * @returns {number} Number of minor currency units per major unit
+ */
+function minorUnitsPerMajorUnitForCurrency( currencyCode ) {
+	const normalizedCurrencyCode = currencyCode.toUpperCase();
+	switch ( normalizedCurrencyCode ) {
+		case 'JPY':
+			return 1;
+
+		case 'AUD':
+		case 'BRL':
+		case 'CAD':
+		case 'EUR':
+		case 'GBP':
+		case 'INR':
+		case 'NZD':
+		case 'USD':
+			return 100;
+
+		default:
+			// Using a default is dangerous here; because it could mean
+			// showing an incorrect price. Better to throw an error.
+			throw 'Unknown currency exponent';
+	}
+}
+
+/**
+ * Formats fractional currency units; different currencies have different customs
+ *
+ * @param {string} currencyCode ISO 4217 currency code
+ * @param {number} absoluteAmount Number of minor currency units
+ * @returns {string}
+ */
+function minorUnitsAsDecimalForCurrency( currencyCode, absoluteAmount ) {
+	const normalizedCurrencyCode = currencyCode.toUpperCase();
+	switch ( normalizedCurrencyCode ) {
+		case 'AUD':
+		case 'CAD':
+		case 'EUR':
+		case 'GBP':
+		case 'INR':
+		case 'NZD':
+		case 'USD':
+			if ( absoluteAmount < 0 || 100 <= absoluteAmount ) {
+				throw '';
+			}
+			return absoluteAmount.toString().padStart( 2, '0' );
+
+		case 'JPY':
+			throw 'Currency does not have a minimal unit: '.concat( normalizedCurrencyCode );
+
+		default:
+			throw 'Minor unit format for currency code not defined: '.concat( normalizedCurrencyCode );
+	}
+}
+
+/**
+ * Graphical currency symbol
+ *
+ * @param {string} currencyCode ISO 4217 currency code
+ * @returns {string} Currency symbol
+ */
+function symbolForCurrency( currencyCode ) {
+	const normalizedCurrencyCode = currencyCode.toUpperCase();
+	switch ( normalizedCurrencyCode ) {
+		case 'AUD':
+		case 'CAD':
+		case 'NZD':
+		case 'USD':
+			return '$';
+
+		case 'GBP':
+			return '£';
+
+		case 'EUR':
+			return '€';
+
+		case 'JPY':
+			return '¥';
+
+		case 'BRL':
+			return 'R$';
+
+		case 'INR':
+			return '₹';
+
+		case 'SEK': // Swedish krona
+		case 'NOK':
+		case 'DKK':
+		case 'ISK': // Icelandic króna
+			return 'kr';
+
+		default:
+			throw 'Symbol for currency code not defined: '.concat( normalizedCurrencyCode );
+	}
+}
+
+/**
+ * Customary decimal and group separators in a given locale.
+ *
+ * @param {string} localeCode ISO 3166-1 alpha-2 region code
+ * @returns {{decimalSeparator:string, groupSeparator:string}} Decimal and group separators
+ */
+function separatorsForLocale( localeCode ) {
```

Comment:
I have no problem with having a custom one, but we could maybe make use of the `Intl.NumberFormat('ja-JP', { style: 'currency', currency: 'JPY' }).format(number)` function to format separators and currency symbols?

Reply:
I didn't know about `Intl.NumberFormat` -- thanks for the pointer.

Just musing about this... I'm torn because on one hand that would handle all of the configuration around separators and grouping conventions, but we'd also lose some flexibility (which is funny because that function takes like 12 options) around currency symbols. For instance, in Australia, should Australian dollars be formatted as `A$ 10` or `A$10` or `$10 AUD`? Does it matter? I don't know. But `toLocaleString` picks this for us.

Also the `NumberFormat` object uses floats, while we're trying to stick with integer arithmetic. I can think of a couple of ways to deal with that. One is to just use a float; divide by the minor unit right before passing the amount to `NumberFormat` and use options to round. This brings us back to the accuracy issue with floats, albeit not nearly as severe because we only divide once at the end by a small integer. It hurts my soul to settle for that when an exact solution is right there, but so it goes. :) Another option is to extract the major unit portion as an integer, format it (as a float), and manually replace the decimal part of the string, but that smells terrible.

Reply:
You make a good case. It's actually a little surprising to me that the built-in methods use floats.  It seems like a big oversight. I suppose the answer is probably that they didn't want to have to include data for major/minor unit conversion. Well, do what you think best! As long as the tests pass we can always change the underlying implementation later.